src/Parallel/ReductionDeclare.hpp

```diff
@@ -0,0 +1,15 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
```

Comment:
This might be one of the few times where a `/// \file` Doxygen comment might be useful. I gather this file exists because including the other one in `ci` files creates very fun cyclic dependencies? Having a comment explaining when to use this file would probably be helpful, since it's not obvious to me right now, and likely less obvious without the context of the rest of this commit :)